iOS/src/toga_iOS/fonts.py

```diff
@@ -47,21 +54,24 @@ def __init__(self, interface):
                 else:
                     family = self.interface.family
 
-                full_name = "{family}{weight}{style}".format(
-                    family=family,
-                    weight=(" " + self.interface.weight.title())
-                    if self.interface.weight is not NORMAL
-                    else "",
-                    style=(" " + self.interface.style.title())
-                    if self.interface.style is not NORMAL
-                    else "",
+                font = UIFont.fontWithName(family, size=size)
+                if font is None:
+                    print(f"Unable to load font: {size}pt {family}")
+                    font = UIFont.systemFontOfSize(size)
+
+            # Convert the base font definition into a font with all the desired traits.
+            traits = 0
+            if self.interface.weight == BOLD:
+                traits |= UIFontDescriptorTraitBold
+            if self.interface.style == ITALIC:
+                traits |= UIFontDescriptorTraitItalic
+            if traits:
+                font = UIFont.fontWithDescriptor(
+                    font.fontDescriptor.fontDescriptorWithSymbolicTraits(traits),
+                    size=size,
                 )
-                font = UIFont.fontWithName(full_name, size=size)
 
-                if font is None:
-                    print(f"Unable to load font: {self.interface.size}pt {full_name}")
-                else:
-                    _FONT_CACHE[self.interface] = font
```

Comment:
As with Cocoa; we either need to report the missing font issue, or docstring the reason why it isn't an issue.